src/main/java/io/stargate/sgv2/jsonapi/service/embedding/configuration/PropertyBasedEmbeddingServiceConfig.java

```diff
@@ -1,40 +1,94 @@
 package io.stargate.sgv2.jsonapi.service.embedding.configuration;
 
+import com.fasterxml.jackson.annotation.JsonProperty;
 import io.smallrye.config.ConfigMapping;
 import io.smallrye.config.WithConverter;
 import io.smallrye.config.WithDefault;
 import jakarta.annotation.Nullable;
-import java.net.URL;
+import java.util.List;
+import java.util.Map;
 import java.util.Optional;
 import org.eclipse.microprofile.config.spi.Converter;
 
-@ConfigMapping(prefix = "stargate.jsonapi.embedding.service")
+@ConfigMapping(prefix = "jsonapi.vector.provider")
 public interface PropertyBasedEmbeddingServiceConfig {
 
   @Nullable
-  ServiceConfig openai();
+  VectorProviderConfig openai();
 
   @Nullable
-  ServiceConfig huggingface();
+  VectorProviderConfig huggingface();
 
   @Nullable
-  ServiceConfig vertexai();
+  VectorProviderConfig vertexai();
 
   @Nullable
-  ServiceConfig cohere();
+  VectorProviderConfig cohere();
 
   @Nullable
-  ServiceConfig nvidia();
+  VectorProviderConfig nvidia();
 
   @Nullable
   CustomConfig custom();
 
-  public interface ServiceConfig {
+  interface VectorProviderConfig {
+    @JsonProperty
     boolean enabled();
 
-    URL url();
+    @JsonProperty
+    String url();
 
+    @JsonProperty
     String apiKey();
+
+    @JsonProperty
+    List<String> supportedAuthentication();
+
+    @Nullable
+    @JsonProperty
+    List<ParameterConfig> parameters();
+
+    @Nullable
+    @JsonProperty
+    Map<String, String> properties();
+
+    @Nullable
+    @JsonProperty
+    List<ModelConfig> models();
+
+    interface ModelConfig {
+      @JsonProperty
+      String name();
+
+      @JsonProperty
+      List<ParameterConfig> parameters();
+
+      @JsonProperty
+      Map<String, String> properties();
+    }
+
+    interface ParameterConfig {
+      @JsonProperty
+      String name();
+
+      @JsonProperty
+      ParameterType type();
+
+      @JsonProperty
+      boolean required();
+
+      @JsonProperty
+      Optional<String> defaultValue();
```

Comment:
Can you mark the fields that can be Nullable

Reply:
Sure, but `Optional` has already made this field Nullable